recipes-ni/ni-base-system-image-tests/files/disk_size.py

```diff
@@ -0,0 +1,89 @@
+import datetime
+import subprocess
+import os
+import argparse
+import sys
+
+TEST_DESC = 'Test to see that the file size of the base system image is not getting too large.'
+
+BASE_YEAR = 2023
+NEXT_YEAR = datetime.datetime.now().year + 1
+
+# Get the max image size for the end of the year (i.e. next year) for ARM
+ARM_BASE_SIZE = 287         # Image size as of writing. Update when changing ARM device
+ARM_SMALLEST_DRIVE = 512    # The size of cRIO ___'s hardrive
+ARM_EOL_YEAR = 2038         # When the cRIO ___ goes EOL
+ARM_INC_RATE = (ARM_SMALLEST_DRIVE - ARM_BASE_SIZE) / (ARM_EOL_YEAR - BASE_YEAR)
+ARM_MAX_SIZE_EOY = (ARM_INC_RATE * (NEXT_YEAR - BASE_YEAR)) + ARM_BASE_SIZE
+
+# Same thing, but for x86
+X86_BASE_SIZE = 1138        # Image size as of writing. Update when changing x86 device
+X86_SMALLEST_DRIVE = 4096   # The size of cRIO ___'s hardrive
+X86_EOL_YEAR = 2038         # When the cRIO ___ goes EOL
+X86_INC_RATE = (X86_SMALLEST_DRIVE - X86_BASE_SIZE) / (X86_EOL_YEAR - BASE_YEAR)
+X86_MAX_SIZE_EOY = (X86_INC_RATE * (NEXT_YEAR - BASE_YEAR)) + X86_BASE_SIZE
```

Comment:
I know you replace this block later in the patchset, but could you squash down the changes do that there aren't false advertisements of EOL dates in the git history?

Reply:
Good catch